src/Util.js

```diff
@@ -156,3 +156,64 @@ export const getSphericalHarmonicsComponentCountForDegree = (sphericalHarmonicsD
     }
     return 0;
 };
+
+export const nativePromiseWithExtractedComponents = () => {
+    let resolver;
+    let rejecter;
+    const promise = new Promise((resolve, reject) => {
+        resolver = resolve;
+        rejecter = reject;
+    });
+    return {
+        'promise': promise,
+        'resolve': resolver,
+        'reject': rejecter
+    };
+};
+
+export const abortablePromiseWithExtractedComponents = (abortHandler) => {
+    let resolver;
+    let rejecter;
+    if (!abortHandler) {
+        abortHandler = () => {};
+    }
+    const promise = new AbortablePromise((resolve, reject) => {
+        resolver = resolve;
+        rejecter = reject;
+    }, abortHandler);
+    return {
+        'promise': promise,
+        'resolve': resolver,
+        'reject': rejecter
+    };
+};
+
+class Semver {
+    constructor(major, minor, patch) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+    }
+
+    toString() {
+        return `${this.major}_${this.minor}_${this.patch}`;
+    }
+}
+
+export function isIOS() {
+    const ua = navigator.userAgent;
+    return ua.indexOf('iPhone') > 0 || ua.indexOf('iPad') > 0 || ua.indexOf('iPhone') > 0;
```

Comment:
You already checked iPhone

Reply:
Good catch! This is what happens when you just copy a function from the internet :sweat_smile: